File: homework-1/prefixes.py
"""This script writes successive prefixes of its first input argument, 
one per line, starting with the first prefix, which is zero characters long.
"""

import sys

arg = sys.argv[1]

def prefixes (string):
	for x in range(0, len(string) + 1):
		print ''.join(string[:x])

prefixes(arg)

File: homework-3/prefixes.go
package main

import (
	"fmt"
	"os"
	"strings"
)

// Writes successive prefixes of its first input argument, one per line.

func prefixes(s string) {
	pre := strings.Split(s, "")
	for i := 0; i < len(pre) + 1; i++ {
		fmt.Println(strings.Join(pre[:i], ""))
	}
}

func main() {
	prefixes(os.Args[1])
}